src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowPRNCBuilder.java

```diff
@@ -0,0 +1,544 @@
+        Set<String> docIds = new HashSet<String>();
+        String documentId;
+        String financialObjectCode;
+        
+        
+        BigDecimal netPaymentAmount;
+        int currentBatchSize = 0;
+        
+        while (rs.next()) {
+            // Initialize variables for current row.
+            documentId = rs.getString(prncRow.preqDocumentNumber.index);
+            financialObjectCode = rs.getString(prncRow.financialObjectCode.index);
+            netPaymentAmount = rs.getBigDecimal(prncRow.amount.index);
+            
+            // Add doc ID to map if non-blank.
+            if (StringUtils.isNotBlank(documentId)) {
+                docIds.add(documentId);
+            }
+                        
+            
+            // If net payment amount is null, then set to zero
+            if (netPaymentAmount == null) {
+                netPaymentAmount = summary.zeroAmount;
+            } 
+
+            // Perform extra 1099-specific or 1042S-specific setup as needed.
+            doTaxSpecificRowSetup(rs, insertStatement, financialObjectCode, summary);
+            
+            insertStatement.setInt(detailRow.reportYear.index - offset, summary.reportYear);
+            insertStatement.setString(detailRow.documentNumber.index - offset, StringUtils.isNotBlank(documentId) ? documentId : null);
+            insertStatement.setString(detailRow.documentType.index - offset, PurapConstants.PurapDocTypeCodes.PAYMENT_REQUEST_DOCUMENT);
+            insertStatement.setInt(detailRow.financialDocumentLineNumber.index - offset, rs.getInt(prncRow.accountIdentifier.index));
+            insertStatement.setString(detailRow.finObjectCode.index - offset, financialObjectCode);
+            insertStatement.setBigDecimal(detailRow.netPaymentAmount.index - offset, netPaymentAmount);
+            insertStatement.setString(detailRow.documentTitle.index - offset, rs.getString(prncRow.paymentMethodCode.index));
+            insertStatement.setString(detailRow.vendorTaxNumber.index - offset, rs.getString(prncRow.vendorTaxNumber.index));
+            insertStatement.setString(detailRow.payeeId.index - offset, rs.getString(prncRow.preqVendorHeaderGeneratedIdentifier.index) +  "-" + rs.getString(prncRow.preqVendorDetailAssignedIdentifier.index));
+            insertStatement.setString(detailRow.vendorTypeCode.index - offset, rs.getString(prncRow.vendorTypeCode.index));
+            insertStatement.setString(detailRow.vendorOwnershipCode.index - offset, rs.getString(prncRow.vendorOwnershipCode.index));
+            insertStatement.setString(detailRow.vendorOwnershipCategoryCode.index - offset, rs.getString(prncRow.vendorOwnershipCategoryCode.index));
+            insertStatement.setString(detailRow.vendorForeignIndicator.index - offset, rs.getString(prncRow.vendorForeignInd.index));
+            insertStatement.setString(detailRow.nraPaymentIndicator.index - offset, rs.getString(prncRow.vendorForeignInd.index));
```

Comment:
good question. This was supposed to mimic pdp source. In pdp, for PREQs the nraPayment flag is set to be vendorForeignIndicator and that is why I set this field to vendorForeignIndicator here.